### PR TITLE
Update to SI 4.1.x

### PR DESCRIPTION
**Pull request by garyrussell**

### CODE REVIEW

File: spring-integration-dsl-groovy-core/src/main/groovy/org/springframework/integration/dsl/groovy/ClosureInvokingComponents.groovy [hunk 1]
```diff
@@ -215,6 +216,15 @@ class ClosureInvokingChannelInterceptor implements org.springframework.messaging
 		result as Message
 	}
 
+	void afterSendCompletion(Message<?> message, MessageChannel channel, boolean sent, Exception ex) {
+		interceptor.afterSendCompletion
```
Comment (artembilan) [inline spring-integration-dsl-groovy-core/src/main/groovy/org/springframework/integration/dsl/groovy/ClosureInvokingComponents.groovy]: Sorry, I forgot a bit Groovy or just haven't used this technique before, hence I don't understand how it works.
I see other methods here use these `Closure`s like `interceptClosure.call(....)`.
Don't we miss something similar here?
Shouldn't we have a test-case on the matter?

**Comment by garyrussell** [comment]
I have no clue; I just copied the existing code 

/cc @dturanski

**Comment by artembilan** [comment]
Well, as I said. The `ClosureInvokingChannelInterceptorTests` shows us that we aren't correct:
```
  boolean afterSendCompletion
        def channelInterceptor = new ChannelInterceptor(preSend: { payload -> "$payload, world" },
                afterSendCompletion: {m -> afterSendCompletion = true})
.....
assert afterSendCompletion
```
Will fix during merge with other concerns which I won't like

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration-dsl-groovy/commit/0b28821c4f39f919db3e63bbe761f2d3e736908b

**Pull request closed without merging**